titan/partnering.py

```diff
@@ -80,15 +53,54 @@ def assort(eligible_partners, assort_params):
         return None
 
     if params.features.assort_mix:
-        for assort_def in params.assort_mix.values():
-            if getattr(agent, assort_def.attribute) == assort_def.agent_value:
-                eligible = assort(eligible, assort_def)
-
-    random_partner = utils.safe_random_choice(eligible, rand_gen)
+        random_partner = None
+        assort_attrs = get_assort_attrs(params.assort_mix.values(), agent, rand_gen)
+        for partner in utils.safe_shuffle(eligible, rand_gen):
+            if is_assortable(partner, assort_attrs):
+                random_partner = partner
+                break
+    else:
+        random_partner = utils.safe_random_choice(eligible, rand_gen)
 
     return random_partner
 
 
+def is_assortable(agent, assort_attrs):
```

Comment:
The new assorting is follow-able since the logic is clean, but I think it might prove troublesome to understand. Can you add some comments on it?